pkgs/by-name/co/cosmic-session/package.nix

```
@@ -0,0 +1,73 @@
+{
+  lib,
+  fetchFromGitHub,
+  bash,
+  rustPlatform,
+  just,
+  dbus,
+  rust,
+  stdenv,
+  xdg-desktop-portal-cosmic,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "cosmic-session";
+  version = "unstable-2023-12-22";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = pname;
+    rev = "e8a429a2de7e07df66e04b1ef69dc5d98b15f008";
+    sha256 = "sha256-kF+YsW5usxYe8c4ksYrGaoAbZiXvLiFYnGrfCfxwR3U=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "cosmic-notifications-util-0.1.0" = "sha256-GmTT7SFBqReBMe4GcNSym1YhsKtFQ/0hrDcwUqXkaBw=";
+      # added vendor
+      "launch-pad-0.1.0" = "sha256-tnbSJ/GP9GTnLnikJmvb9XrJSgnUnWjadABHF43L1zc=";
+    };
+  };
+
+  postPatch = ''
+    substituteInPlace Justfile --replace '#!/usr/bin/env' "#!$(command -v env)"
+    substituteInPlace Justfile --replace 'target/release/cosmic-session' "target/${
+      rust.lib.toRustTargetSpecShort stdenv.hostPlatform
+    }/release/cosmic-session"
+    substituteInPlace data/start-cosmic --replace '#!/bin/bash' "#!${bash}/bin/bash"
+    substituteInPlace data/start-cosmic --replace '/usr/bin/cosmic-session' "${
+      placeholder "out"
+    }/bin/cosmic-session"
+    substituteInPlace data/start-cosmic --replace '/usr/bin/dbus-run-session' "${dbus}/bin/dbus-run-session"
+    #TODO: should be doable with a session dir variable
+        substituteInPlace data/cosmic.desktop --replace '/usr/bin/start-cosmic' "$out/bin/start-cosmic"
+  '';
+
+  nativeBuildInputs = [ just ];
+  buildInputs = [ ];
+
+  dontUseJustBuild = true;
+
+  justFlags = [
+    "--set"
+    "prefix"
+    (placeholder "out")
+    "--set"
+    "xdp_cosmic"
+    "${xdg-desktop-portal-cosmic}"
```

```suggestion
    xdg-desktop-portal-cosmic
```

No need to make Nix do more work here with string antiquotation